src/test/java/bio/terra/workspace/service/datareference/model/ReferenceObjectTest.java

```diff
@@ -0,0 +1,34 @@
+package bio.terra.workspace.service.datareference.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import org.junit.jupiter.api.Test;
+
+public class ReferenceObjectTest extends BaseUnitTest {
+
+  /**
+   * This is a simple test for verifying that SnapshotReference's toJson() and fromJson() are
+   * compatible. If this failing, you've likely broken one or both of those methods.
+   */
+  @Test
+  public void SnapshotReferenceSerializationWorks() {
+    SnapshotReference snapshot =
+        SnapshotReference.create(/* instanceName= */ "foo", /* snapshot= */ "bar");
+    assertEquals(snapshot, ReferenceObject.fromJson(snapshot.toJson()));
+  }
+
+  /**
+   * Hard code serialized values to check that code changes do not break backwards compatibility of
+   * stored JSON values. If this test fails, your change may not work with existing databases.
+   */
+  @Test
+  public void SnapshotReferenceSerializationBackwardsCompatible() {
+    SnapshotReference snapshotReference =
+        (SnapshotReference)
+            ReferenceObject.fromJson(
+                "{\"@type\":\"AutoValue_SnapshotReference\",\"instanceName\":\"foo\",\"snapshot\":\"bar\"}");
```

Comment:
there is weirdness with auto value object mapper transformations. I think we would be better off doing the serialization manually. As-is, any class name change is going to necessitate backwards compatibility code. that's not a great place to be. That's some boilerplate, but it's fine to have 1 place of serialization boilerplate per class.

Reply:
I think an easier way to avoid that issue is to explicitly set the type name we want to use instead of using the class name, which is straightforward with the annotation `JsonTypeName` annotation. I used `SnapshotReference` as the value here for clarity, but I also tested it with arbitrary strings, which work fine. I'd prefer to avoid manual serialization if possible, since it's a non-trivial amount of boilerplate to maintain